File: leet_simplify_path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        ss = []
        for s in path.split('/') :
            if s and s != '.' :
                if s == ".." :
                    if ss :
                        ss.pop()
                else :
                    ss.append(s)
        return '/' + '/'.join(ss)
    
                    

File: LearnJava/Leet_simplify_path.java
class Leet_simplify_path {
    public String simplifyPath(String path) {
        Stack<String> ss = new Stack<>();
        
        for (String s : path.split("/")) {
            if (s.equals(".") || s.equals("")) {
                continue;
            }
            
            if (s.equals("..")) {
                if (!ss.empty()) {
                    ss.pop();
                }
            }
            else {
                ss.push(s);
            }
        }
        
        return '/' + String.join("/", ss);
    }
}